### ISSUE TITLE
WSM: Decay-Free Learning Rate Schedule via Checkpoint Merging for LLM Pre-training

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#393]
## Links
https://arxiv.org/abs/2507.17634

## TLDR

本研究は、学習率(LR)減衰フェーズをチェックポイントマージングで置き換える新しいLLM事前学習フレームワークWSMを提案する。WSMは様々な減衰戦略を理論的にエミュレートし、継続的な学習を可能にしながら、従来のWSD手法を大幅に上回る性能を達成した。

This paper proposes WSM, a novel LLM pre-training framework that replaces the learning rate (LR) decay phase with checkpoint merging. WSM theoretically emulates various decay strategies, enabling continual training while achieving significantly superior performance over the conventional WSD method.

### 1. はじめに：WSMフレームワークの提案
本研究は、大規模言語モデル（LLM）の事前学習における学習率（LR）スケジューリングの課題に取り組むものです。従来のLRスケジュール（例：コサイン減衰）は、総学習ステップ数を事前に決める必要があり、学習の延長が困難でした。この問題を緩和する**WSD（Warmup-Stable-Decay）**も、減衰フェーズの設計が手動で柔軟性に欠けるという課題が残ります。
そこで本論文は、この減衰フェーズを完全に排除し、代わりにチェックポイントのマージングでその効果を代替する新しいフレームワーク**WSM (Warmup-Stable and Merge)**を提案します。WSMでは、学習自体はWarmup後にLRを一定に保つStableフェーズのみで継続し、その間に保存された複数のチェックポイントを後から統合（マージ）することで、LR減衰を適用したかのような高性能なモデルを生成します。

### 2. これまでの研究との違いや新規性
WSMの新規性は、以下の2点に集約されます。

- LR減衰とチェックポイントマージングの理論的等価性の確立:
本研究は、チェックポイントの加重平均（マージ）が、学習開始点からの勾配更新に対して特定の減衰係数を適用することと数学的に等価であることを証明しました（Theorem 3.1）。これにより、単なる経験則ではなく、理論的な裏付けを持って様々なLR減衰曲線（コサイン、線形、逆平方根など）をマージ重みの設計によってエミュレート（模倣）することが可能になりました。

- 減衰フェーズの完全な排除による柔軟性の獲得:
先行研究のWSDでは必須だった減衰フェーズを完全になくしました。これにより、学習をいつまでも中断・延長でき、必要に応じて後から好きな減衰戦略を「マージ」という形で適用できるという、極めて高い柔軟性を実現しました。これは、特に継続的なデータ追加や長期的なモデル改善が求められるLLM開発において大きな利点となります。

### 3. Contributionの実証方法
論文の主張は、広範な実験と分析によって実証されています。

- 性能比較実験:
16.3億パラメータのMoE（Mixture-of-Experts）モデルを用い、ベースラインであるWSD手法とWSMの性能を多数のベンチマークで比較しました。その結果、WSMはWSDを大幅に上回り、特に**MATHで+3.5%、HumanEvalで+2.9%、MMLU-Proで+5.5%**という顕著な性能向上を達成しました。さらに、この性能優位性は、教師ありファインチューニング（SFT）後も持続することが確認されています。

- 要因分析:
マージングの性能に影響を与える要因（マージアルゴリズム、チェックポイントの保存間隔、マージする数など）を体系的に調査しました。その結果、マージ対象のチェックポイントがカバーする学習期間である**「マージ期間（merge duration）」**が最も重要な要因であることを突き止めました。

- 理論の検証:
理論的に性能が良いとされる凹型の減衰曲線（逆平方根減衰）を模倣したマージアルゴリズムが、線形（単純平均）や凸型（EMA）を模倣したものより優れた性能を示すことを実験的に確認し、理論の正しさを裏付けました。

### 4. 重要なFigureに関する説明

- Figure 1: WSMと主流のLRスケジュールの比較
この図は、WSMのコンセプトを視覚的に示しています。従来のコサイン減衰やWSDが、グラフの右側でLRが低下する「減衰フェーズ」を持つのに対し、WSMはLRが一定のまま学習を継続します。減衰の効果は、破線で示されるように、保存されたチェックポイントのマージによって後からシミュレートされます。これにより、訓練をいつでも延長できる柔軟性が生まれます。

<img width="926" height="329" alt="Image" src="https://github.com/user-attachments/assets/8c31a645-4128-47f6-bfbf-8c321249b8db" />

- Figure 2: チェックポイントマージ重みと減衰曲線の対応
この図は、本研究の理論的貢献の中核をなす「マージ重みと減衰曲線の対応関係」を可視化したものです。例えば、(b)線形減衰は全てのチェックポイントを均等に重み付け（単純平均）することで、(c)コサイン減衰や(d)逆平方根（1-sqrt）減衰は、より最近のチェックポイントに大きな重みを置く非線形な重み付けでエミュレートできることを示しています。

<img width="1144" height="634" alt="Image" src="https://github.com/user-attachments/assets/72c8a6ef-2926-4d31-b29b-3d165b7617bb" />

- Figure 6: パラメータ軌跡の可視化
この図は、モデルのパラメータ空間をt-SNEで2次元に圧縮し、性能を等高線で示したものです。定数LRでの学習軌跡（constant、青い点）は振動しながら進むのに対し、LR減衰の軌跡（Decay-8T、オレンジの点）はより滑らかに収束します。そして、これらの軌跡が最終的に向かう先にある**マージモデル（Merged-Model、赤い星）**が、最も性能の高い領域（等高線の中心）に位置していることがわかります。これは、マージがより安定した最適化経路を提供し、高性能な解を発見するのに有効であることを示唆しています。

<img width="864" height="612" alt="Image" src="https://github.com/user-attachments/assets/92d06510-6252-4045-844e-3ca130471cac" />
